lib/datastore/entity.js

```diff
@@ -293,14 +308,17 @@ module.exports.formatArray = formatArray;
  */
 module.exports.isKeyComplete = function(key) {
```

Comment:
This doesn't quite fit Datastore's notion of complete vs incomplete keys. In particular, a key is incomplete if the last element has no id or name. An unset kind or an ancestor path element with no id or name is just an invalid key.